Stgwotempvar

sridharxp opened the pull request:
[@pboyer](https://github.com/pboyer), [@millergarym](https://github.com/millergarym)
Creating Temp Var _x looks to me error prone.
Following java.stg seems to me more robust.
In this way Var re declaration issue does not arise.
For [_x redeclared in this block](https://github.com/pboyer/antlr4/issues/88) this seems to be a safer solution than the PR [Fix for redeclared error is generated Go code](https://github.com/pboyer/antlr4/pull/89)

Congratulations to Antlr team and specially to Go contributors

pboyer commented:
So long as the tests do not regress. LGTM. Thanks @sridharxp.

parrt commented:
looks like we are getting some errors in the go target:

```
...
Running org.antlr.v4.test.runtime.go.TestPerformance
Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 1.909 sec <<< FAILURE!
testOne[Go:ExpressionGrammar_2](org.antlr.v4.test.runtime.go.TestPerformance)  Time elapsed: 0.159 sec  <<< FAILURE!
junit.framework.ComparisonFailure: expected:<null> but was:<# _/tmp/BaseGoTest-1479284708117/parser
parser/expr_parser.go:194: too many arguments in call to p.Expr
parser/expr_parser.go:338: too many arguments in call to p.Expr
parser/expr_parser.go:371: too many arguments in call to p.Expr
parser/expr_parser.go:384: too many arguments in call to p.Expr
...
```

sridharxp commented:
[@pboyer](https://github.com/pboyer)
Found the bug; Instance of capitalizing convention applied to wrong identifier. 
Should I close and raise a separate PR?

sridharxp commented:
[@pboyer](https://github.com/pboyer)
Corrected `TestPerformance,java` ; Instance of capitalizing convention applied to wrong identifier. 
Meanwhile found some more failures are in `TestLeftRecursion.java`...
Updated code
```
InvokeRule(r, argExprsChunks) ::= <<
p.SetState(<r.stateNumber>)
<if(r.labels)><r.labels:{l | <labelref(l)> = }><endif><if(r.ast.options.p)>p.<r.name>(<r.ast.options.p><if(argExprsChunks)>,<endif><else>p.<r.name; format="cap">(<endif><argExprsChunks>);
>>
MatchToken(m) ::= <<
p.SetState(<m.stateNumber>)
<if(m.labels)><m.labels:{l | <labelref(l)> = }><endif>p.Match(<parser.name><m.name>);
>>
```

parrt commented:
This breaks everything. apparently go doesn't like multiple `=`:

```
localctx.(*SContext).q = localctx.(*SContext)._e = p.e(0);
```

pboyer commented:
Reverting. @sridharxp this breaks various tests. You'll need to look at the travis output for details. If possible, please verify there are no regressions before submitting a PR by running the full test suite (e.g. `mvn install`)

pboyer commented:
@parrt It looks like I can't revert the "easy" way. I'll need to take a look when I get home.

parrt commented:
I fixed master. have no fear :)

pboyer commented:
Phew!
